### PR TITLE
remove data namespace when pushing data to state object

### PR DESCRIPTION
**Pull request by bpauli**
- adapt unit test

### CODE REVIEW

File: examples/js/datalayer.mocks.1.js [hunk 1]
```diff
@@ -427,13 +419,12 @@ governing permissions and limitations under the License.
       // -----------------------------------------------------------------------------------------------------------------
 
       adobeDataLayer.push({
-        data: {
```
Comment (jckautzmann) [inline examples/js/datalayer.mocks.1.js:430]: this test does not make sense anymore as there is no invalid format: I'd remove it.

File: examples/js/datalayer.mocks.1.js [hunk 2]
```diff
@@ -447,12 +438,11 @@ governing permissions and limitations under the License.
       adobeDataLayer.push({
         event: 'clicked',
         data: {
-          invalid: {}
```
Comment (jckautzmann) [inline examples/js/datalayer.mocks.1.js:450]: this test does not make sense anymore as there is no invalid format: I'd remove it.

File: src/scripts/DataLayerItem.js [hunk 3]
```diff
@@ -89,17 +83,22 @@ class Item {
...
         type = DataLayer.constants.itemType.LISTENER_OFF;
+      } else if (isPlainObject(config)) {
+        type = DataLayer.constants.itemType.DATA;
       }
       return type;
     }(itemConfig));
+
+    this._config.data = (function(config, type) {
+      return omit(config, Object.keys(constraints[type + 'Config']));
+    })(itemConfig, this._type);
```
Comment (jckautzmann) [inline src/scripts/DataLayerItem.js:100]: What's the intention of this logic?
Couldn't it be simplified as follows?
`this._config.data = omit(itemConfig, Object.keys(constraints[this._type + 'Config']));`

File: src/tests/DataLayer.test.js [hunk 4]
```diff
@@ -417,64 +424,56 @@ describe('listener with path', () => {
       handler: mockCallback
     };
     dataLayer.push(listenerOn);
-    dataLayer.push({
+    dataLayer.push(Object.assign({
       event: 'viewed',
-      data: carouselData
-    });
```
Comment (jckautzmann) [inline src/tests/DataLayer.test.js:423]: I'd simplify as follows:
```
    dataLayer.push({
      event: 'viewed',
      carouselData
    });
```

File: src/tests/DataLayer.test.js [hunk 5]
```diff
@@ -417,64 +424,56 @@ describe('listener with path', () => {
...
+
+    }, carouselData));
     expect(mockCallback.mock.calls.length).toBe(0);
-    dataLayer.push({
+    dataLayer.push(Object.assign({
```
Comment (jckautzmann) [inline src/tests/DataLayer.test.js:432]: see above

File: src/tests/DataLayer.test.js [hunk 6]
```diff
@@ -417,64 +424,56 @@ describe('listener with path', () => {
...
-      data: imageData
-    });
+    }, imageData));
     expect(mockCallback.mock.calls.length).toBe(1);
   });
 
   test('listener on: datalayer:change with path', () => {
     dataLayer.push(listenerOn);
-    dataLayer.push({
+    dataLayer.push(Object.assign({
```
Comment (jckautzmann) [inline src/tests/DataLayer.test.js:440]: see above

File: src/tests/DataLayer.test.js [hunk 7]
```diff
@@ -417,64 +424,56 @@ describe('listener with path', () => {
...
       event: 'datalayer:change',
-      data: carouselData
-    });
+    }, carouselData));
     expect(mockCallback.mock.calls.length).toBe(0);
-    dataLayer.push({
+    dataLayer.push(Object.assign({
```
Comment (jckautzmann) [inline src/tests/DataLayer.test.js:444]: see above

File: src/tests/DataLayer.test.js [hunk 8]
```diff
@@ -417,64 +424,56 @@ describe('listener with path', () => {
...
   test('listener: custom event, path and scope: all', () => {
-    dataLayer.push({
-      event: 'datalayer:change',
-      data: carouselData
-    });
-    dataLayer.push({
-      event: 'datalayer:change',
-      data: imageData
-    });
+    dataLayer.push(Object.assign({
```
Comment (jckautzmann) [inline src/tests/DataLayer.test.js:451]: see above

File: src/tests/DataLayer.test.js [hunk 9]
```diff
@@ -417,64 +424,56 @@ describe('listener with path', () => {
...
+      event: 'datalayer:change'
+    }, carouselData));
+    dataLayer.push(Object.assign({
```
Comment (jckautzmann) [inline src/tests/DataLayer.test.js:454]: see above

File: src/tests/DataLayer.test.js [hunk 10]
```diff
@@ -417,64 +424,56 @@ describe('listener with path', () => {
...
+      event: 'datalayer:change'
+    }, imageData));
     dataLayer.push({
       on: 'datalayer:change',
       path: 'component',
       scope: 'all',
       handler: mockCallback
     });
-    dataLayer.push({
+    dataLayer.push(Object.assign({
```
Comment (jckautzmann) [inline src/tests/DataLayer.test.js:463]: see above

File: src/tests/DataLayer.test.js [hunk 11]
```diff
@@ -528,10 +525,9 @@ describe('listener with path', () => {
         }
       }
     };
-    dataLayer.push({
-      event: 'datalayer:change',
-      data: oldData
-    });
+    dataLayer.push(Object.assign({
```
Comment (jckautzmann) [inline src/tests/DataLayer.test.js:528]: see above

File: src/tests/DataLayer.test.js [hunk 12]
```diff
@@ -543,10 +539,9 @@ describe('listener with path', () => {
         }
       }
     });
-    dataLayer.push({
+    dataLayer.push(Object.assign({
```
Comment (jckautzmann) [inline src/tests/DataLayer.test.js:542]: see above

File: src/tests/DataLayer.test.js [hunk 13]
```diff
@@ -718,60 +711,6 @@ test('listener off: unregister multiple handlers', () => {
...
-  };
-  dataLayer.push({
-    event: 'clicked',
-    data: data,
-    invalid: 'invalid'
-  });
-  expect(dataLayer.getState()).toStrictEqual({});
-});
-
-test('invalid listener on', () => {
```
Comment (jckautzmann) [inline src/tests/DataLayer.test.js:758]: I'd keep the code for invalid listeners until we update the logic with add/remove event listeners.

File: src/tests/DataLayer.test.js [hunk 14]
```diff
@@ -791,106 +730,6 @@ test('invalid listener on scope', () => {
...
-    off: 'carousel 14 clicked',
-  });
-  expect(dataLayer.length).toStrictEqual(0);
-});
-
-// -----------------------------------------------------------------------------------------------------------------
-// getState()
-// -----------------------------------------------------------------------------------------------------------------
-
-test('getState()', () => {
```
Comment (jckautzmann) [inline src/tests/DataLayer.test.js:876]: this test should not be removed: it's testing the logic I added yesterday :)

File: src/tests/DataLayer.test.js (hunk 13 above)
Comment (bpauli) [reply src/tests/DataLayer.test.js:758]: I added the unit test back again but skip it for now.

**Pull request merged**